### PR TITLE
Fixes #684

### LINKED ISSUE #684: isJsonPreferred() must be replaced by ContentNegotationResolver
In `FormController#onErrorSubmit` there is a line doing 

```
if (UserAgents.get(request).isJsonPreferred()) {
``` 

This does not seem right, we should be using the `ContentNegotationResolver` in order to know the kind of expected response.

There might be other places where `isJsonPreferred ` or `isHtmlPreferred()` is used. Please search and improve them all.

### PR DESCRIPTION
**Pull request by jarias**
isJsonPreferred() must be replaced by ContentNegotationResolver

### CODE REVIEW

**Comment by mraible** [comment]
@jarias Should `DefaultUserAgent#isJsonPreferred` be deprecated, along with `DefaultUserAgent#isHtmlPreferred`?

**Comment by jarias** [comment]
@mraible no cause its use by DefaultContentNegotiationResolver which is the proper class to use for resolving the content type

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java [hunk 1]
```diff
@@ -206,7 +205,7 @@ protected ViewModel onErrorSubmit(HttpServletRequest request, HttpServletRespons
 
         List<ErrorModel> errors = toErrors(request, form, e);
 
-        if (UserAgents.get(request).isJsonPreferred()) {
+        if (isJsonPreferred(request, response)) {
```
Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java]: Inside `isJsonPreferred(...)` I see that `ContentNegotiationResolver.INSTANCE` is executed every time. We could simply have a `ContentNegotiation` instance as a property of this class which is used there. Could you please change that? Thanks

**Comment by mrioan** [comment]
@jarias. I just added a comment. Back to you. Thanks

**Comment by jarias** [comment]
@mrioan PR updated

**Comment by mraible** [comment]
LGTM 👍

**Pull request merged**